### PR TITLE
New message tracing events updated

### PR DESCRIPTION
**Pull request by AnujaLK**
JIRA - https://wso2.org/jira/browse/MB-1152

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/delivery/DisruptorBasedFlusher.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public DisruptorBasedFlusher() {
     public void submit(LocalSubscription subscription, AndesMessageMetadata metadata) {
 
         //Tracing Message
-        MessageTracer.trace(metadata, MessageTracer.PUBLISHED_TO_DISRUPTOR);
+        MessageTracer.trace(metadata, MessageTracer.PUBLISHED_TO_OB_DISRUPTOR);
```
Comment (hastef88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/delivery/DisruptorBasedFlusher.java]: OB = OutBound ? Would be clearer if mentioned in the comment line :)

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 2]
```diff
@@ -31,33 +31,48 @@
...
-	/**
-	 * This method will print debug logs for message activities. This will accept andes message as
-	 * a parameter
-	 * @param message Andesmessage
-	 * @param content Message activity
-	 */
-    public static void trace(AndesMessage message, String content) {
+
+
+    public static void trace(long messageId, String destination, String content) {
```
Comment (pumudu88) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: add method comments

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 3]
```diff
@@ -31,33 +31,53 @@
...
+     */
+    public static void trace(long messageId, String destination, String content) {
         if(log.isTraceEnabled()) {
             String messageContent = "Message {";
-            messageContent += " Destination: " + message.getMetadata().getDestination();
-            if (message.getMetadata().getMessageID() > 0) {
-                messageContent += " , Id: " + message.getMetadata().getMessageID();
+            messageContent += " Destination: " + destination;
+            if (messageId > 0) {
+                messageContent += " , Id: " + messageId;
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: += is discouraged.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 4]
```diff
@@ -31,33 +31,53 @@
...
             }
             messageContent += " } ";
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: change this too.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java (hunk 3 above)
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: Its ideal if you use StringBuilder here.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 5]
```diff
@@ -65,15 +85,9 @@ public static void trace(AndesMessage message, String content) {
...
-            messageContent += " Destination: " + metadata.getDestination();
-            if (metadata.getMessageID() > 0) {
-                messageContent += " , Id: " + metadata.getMessageID();
-            }
-            messageContent += " } ";
-            log.trace(messageContent + content);
-        }
+        trace(metadata.getMessageID(),
+                   metadata.getStorageQueueName(),
+                   content);
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: Shall we make this into one line ?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java [hunk 6]
```diff
@@ -269,6 +270,14 @@ public void deleteMessages(final String storageQueueName,
             throws AndesException {
         try {
             wrappedInstance.deleteMessages(storageQueueName, messagesToRemove, deleteAllMetaData);
+
+            //Tracing message activity
+            if(MessageTracer.isEnabled()) {
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java]: is this correct indentation?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java [hunk 7]
```diff
@@ -269,6 +270,14 @@ public void deleteMessages(final String storageQueueName,
...
+                for(long messageId : messagesToRemove) {
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java]: check for indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 8]
```diff
@@ -31,31 +31,52 @@
...
-	/**
+    /**
+     * This method will print debug logs for message activities. This will accept parameters for
+     * message id, destination name and activity message
+     * @param messageId Andes message id
+     * @param destination destination name
+     * @param content message activity
+     */
+    public static void trace(long messageId, String destination, String content) {
+        if(log.isTraceEnabled()) {
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: indentation seems to be wrong. (need a space here)

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java (hunk 6 above)
Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java]: Checked. There is no issue with the indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java (hunk 7 above)
Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingMessageStore.java]: Checked. There is no issue with the indentation.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundEventManager.java [hunk 9]
```diff
@@ -204,6 +208,10 @@ public void ackReceived(AndesAckData ackData) {
         // make the event available to EventProcessors
         ringBuffer.publish(sequence);
 
+        //Tracing message
+        MessageTracer.trace(ackData.getMessageID(), ackData.getDestination(),
```
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundEventManager.java]: Check whether tracing is enabled or not

Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundEventManager.java]: Fixed with PR https://github.com/wso2/andes/pull/257/

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 10]
```diff
@@ -65,15 +86,7 @@ public static void trace(AndesMessage message, String content) {
...
-        if(log.isTraceEnabled()) {
-            String messageContent = "Message {";
-            messageContent += " Destination: " + metadata.getDestination();
-            if (metadata.getMessageID() > 0) {
-                messageContent += " , Id: " + metadata.getMessageID();
-            }
-            messageContent += " } ";
-            log.trace(messageContent + content);
-        }
+        trace(metadata.getMessageID(), metadata.getStorageQueueName(), content);
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: Why do you use getStorageQueueName() here and getDestination() in the previous method? Shouldn't we use the same thing in both cases?

Comment (AnujaLK) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: Fixed with PR https://github.com/wso2/andes/pull/276

File: modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java [hunk 11]
```diff
@@ -31,31 +31,52 @@
...
+    /**
+     * This method will print debug logs for message activities. This will accept parameters for
+     * message id, destination name and activity message
+     * @param messageId Andes message id
+     * @param destination destination name
+     * @param content message activity
+     */
+    public static void trace(long messageId, String destination, String content) {
+        if (log.isTraceEnabled()) {
+	        StringBuilder messageContent = new StringBuilder();
```
Comment (hemikak) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/tools/utils/MessageTracer.java]: +1 for using StringBuilder

**Pull request merged**